npm/src/db/sql/sql.ts

```diff
@@ -140,13 +140,26 @@ class Sql implements DatabaseDriver {
     return JSON.parse(JSON.stringify(response)) || [];
   }
 
-  async getByIndex(namespace: string, idx: Index): Promise<any> {
-    const res = await this.indexRepository.findBy({
-      key: dbutils.keyForIndex(namespace, idx),
-    });
-
+  async getByIndex(namespace: string, idx: Index, pageOffset?: number, pageLimit?: number): Promise<any> {
+    const offsetAndLimitValueCheck = !dbutils.isNumeric(pageOffset) && !dbutils.isNumeric(pageLimit);
+    const skip = Number(offsetAndLimitValueCheck ? 0 : pageOffset);
+
+    let take = Number(offsetAndLimitValueCheck ? this.options.pageLimit : pageLimit);
+
+    take += skip;
```

Comment:
This doesn't look right, why is skip being added to take? I think take is how many you want and skip is how many to skip